AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m

```diff
@@ -245,20 +339,55 @@ - (BOOL)setupAudioEngine {
 
 - (void)teardownAudioEngine {
     [self teardownPlayer];
-    [_engine stop];
-    _engine = nil;
+    [_playoutEngine stop];
+    _playoutEngine = nil;
+
+    [_recordEngine stop];
+    _recordEngine = nil;
 }
 
 - (void)playMusic {
-    if (!_engine) {
-        NSLog(@"Cannot play music. AudioEngine has not been created yet.");
+    NSString *fileName = [NSString stringWithFormat:@"%@/%@", [[NSBundle mainBundle] bundlePath], @"mixLoop.caf"];
+    NSURL *url = [NSURL fileURLWithPath:fileName];
+    AVAudioFile *file = [[AVAudioFile alloc] initForReading:url error:nil];
+
+    AVAudioPCMBuffer *buffer = [[AVAudioPCMBuffer alloc] initWithPCMFormat:file.processingFormat
+                                                             frameCapacity:(AVAudioFrameCount)file.length];
+    NSError *error = nil;
+    BOOL success = [file readIntoBuffer:buffer error:&error];
```

Comment:
Should this operation be done asynchronously (not on the calling thread)?

Reply:
As discussed lets leave a comment saying if its a big file ...